lib/cambiatus_web/resolvers/shop.ex

```diff
@@ -5,6 +5,36 @@ defmodule CambiatusWeb.Resolvers.Shop do
 
   alias Cambiatus.Shop
 
+  def upsert_product(_, %{id: product_id} = params, %{context: %{current_user: current_user}}) do
+    params = Map.merge(params, %{creator_id: current_user.account})
+
+    with product <- Shop.get_product(product_id),
+         {:ok, updated_product} <- Shop.update_product(product, params) do
+      {:ok, updated_product}
+    else
+      nil ->
+        {:error, "Product not found", details: nil}
+
+      {:error, error} ->
+        Sentry.capture_message("Product update failed", extra: %{error: error})
+        {:error, message: "Product update failed", details: Cambiatus.Error.from(error)}
+    end
+  end
+
+  def upsert_product(_, params, %{context: %{current_user: current_user}}) do
+    params
+    |> Map.merge(%{creator_id: current_user.account})
+    |> Shop.create_product()
+    |> case do
+      {:error, reason} ->
+        Sentry.capture_message("Product update failed", extra: %{error: reason})
+        {:error, message: "Product update failed", details: Cambiatus.Error.from(reason)}
+
+      {:ok, product} = result ->
```

Comment:
There's a warning here that `product` is unused
```suggestion
      {:ok, _product} = result ->
```